### DISCUSSION TITLE
MQTT commands being ignored

### DISCUSSION THREAD

**Discussion by arhunt35** [arendst/Tasmota#10228] [Q&A] [answered]
Hi folks.  I am running Tasmota 9.2.0 on an Avatar smart plug.  The web interface works perfectly to toggle the smart plug and the device also connects to my MQTT broker and publishes status etc.  However, it will not respond to any commands sent over MQTT.  It feels like it isn't subscribing to the cmnd topic, although the console log claims it is.  I have tried flashing earlier firmware but that hasn't helped.  I have another identical plug which doesn't have the same issue.  The plug has a unique MQTT client name etc.  Could this be a hardware issue?  Any suggestions?

**Comment by sfromis** [comment, marked as answer]
If you've seen it subscribing to the cmnd topic in the log, the most probable is that you have some sort of "minor" detail, like a typo, or use characters treated differently from what you want. In general, stick to a-z, 0-9, - and _

**Reply by arhunt35** [reply]
Thanks for your reply.  I've used auto discover for Home Assistant so am not manually publishing to cmnd (although I've tried that as well).  I'm pretty sure it's not a typo as I've copied the same config file from the working plug.

**Comment by sfromis** [comment]
Did you look into the last part of my message about characters in the topic? Also try with `weblog 3` checking messages in the Tasmota console when you expect action, and also to recheck what topics Tasmota is exactly subscribing to (visible right after restart).

**Reply by arhunt35** [reply]
Thanks again.  I set up a new mqtt user with no special characters but I get the same behaviour.  The list of subscriptions comes up as:

```
12:23:01.891 MQT: Attempting connection...
12:23:01.928 MQT: Connected
12:23:01.933 MQT: smart_plug_new/tele/LWT = Online (retained)
12:23:01.936 MQT: smart_plug_new/cmnd/POWER = 
12:23:01.939 MQT: Subscribe to smart_plug_new/cmnd/#
12:23:01.944 MQT: Subscribe to tasmotas/cmnd/#
12:23:01.947 MQT: Subscribe to cmnd/DVES_C384A4_fb/#
12:23:01.951 MQT: Subscribe to homeassistant/status
```

I can't get a response on any of the cmnd topics.  The device works as an emulated Hue with Alexa but no luck over MQTT.

**Comment by arendst** [comment]
From your subscribed topics I guess your ``fulltopic`` and ``topic`` are wrong. Pls provide them here.

**Reply by arhunt35** [reply]
Thanks Theo.   I have copied below.

"Topic":"smart_plug_new"
"FullTopic":"%topic%/%prefix%/"

**Comment by arendst** [comment]
OK. So what is the exact MQTT command line you try to send?

**Reply by arhunt35** [reply]
I've tried publishing ON, OFF and TOGGLE to smart_plug_new/cmnd/POWER .  The same commands work with my other Tasmota devices.

**Comment by sfromis** [comment]
Again, try with `weblog 3` checking messages in the Tasmota console when you expect action.

**Reply by arhunt35** [reply]
Thanks.  I've been trying that but nothing comes up in the console when the command is published so it hasn't given me much more to go on.

**Comment by Frogmore42** [comment]
This is a problem best handled on the realtime discord support channel.  You probably have something slightly misconfigured.  It will require some troubleshooting to figure out exactly what.

**Comment by arhunt35** [comment]
Hi all.  I just wanted to say thanks for your help with this.  I'm not sure what fixed it in the end but it started working again, seemingly without any setting changes (although obviously something must have changed).

**Comment by sim0njo** [comment]
Hi,
I am having the same issue running Tasmota Zigbee 9.2.0.3 on a Sonoff Zigbee Gateway.

Paired some Aqara Opple/IKEA buttons and a Tuya powerplug successful, I can see them report button pushes and powerplug state in the console and over MQTT, but I cannot send any commands via MQTT to them.

Working from console allows me to flip the powerplug, so the device and the Zigbee comms are ok.

When tracing Tasmota communication with the MQTT broker, I see NO subscribe being done by Tasmota (although the console reports it), and I tried to determine from this thread what the problem could be, but there is no real solution defined here.

My 'Topic' = 'avdln2/z2m/tzb' and 'Full Topic' = 'avdln2/z2m/tzb/%prefix%'

I expect Tasmota to respond to Topic='avdln2/z2m/tzb/#' or 'avdln2/z2m/tzb/cmnd'

Any idea what is the cause

**Reply by sim0njo** [reply]
In the meantime I did some more debugging... seems like Tasmota is sending multiple MQTT msgs in 1 TCP packet, while the broker assumed that 1 TCP packet == 1 MQTT msg.

So time to fix the MQTT broker, case closed.
